website/docs/migration/family_and_dispose/family_and_dispose.dart

```diff
@@ -0,0 +1,26 @@
+import 'dart:math';
+
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+import '../utils.dart';
+
+part 'family_and_dispose.g.dart';
+
+/* SNIPPET START */
+@riverpod
+class BugsEncounteredNotifier extends _$BugsEncounteredNotifier {
+  late String _id;
+  @override
+  FutureOr<int> build(String featureId) {
+    _id = featureId;
+    return 99;
+  }
+
+  Future<void> fix(int amount) async {
+    state = await AsyncValue.guard(() async {
```

Comment:
That's not the recommended way to deal with side-effects.
I don't suggest storing errors in the state.

Reply:
This is supposed to be a mutation, not a side effect.